integration-provisioning/src/test/java/org/symphonyoss/integration/provisioning/service/KeyPairServiceTest.java

```diff
@@ -0,0 +1,266 @@
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Unit test for {@link KeyPairService}
+ * Created by rsanchez on 14/06/17.
+ */
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore("javax.management.*")
+@PrepareForTest({ KeyPairService.class })
+public class KeyPairServiceTest {
+
+  private static final String MOCK_APP_TYPE = "appTest";
+
+  private static final String MOCK_APP_ID = "mock";
+
+  private static final String MOCK_KEY_PASSWORD = "changeit";
+
+  private static final String MOCK_KEY_FILE = "mock-key.pem";
+
+  private static final String MOCK_CERT_FILE = "mock-cert.pem";
+
+  private static final String MOCK_CERT_CA_CHAIN_FILE = "mock-chain.pem";
+
+  @Mock
+  private ApplicationArguments arguments;
+
+  @Mock
+  private Runtime runtime;
+
+  @Mock
+  private Process process;
+
+  @Mock
+  private UserPrincipalLookupService lookupService;
+
+  @Mock
+  private PosixFileAttributeView attributeView;
+
+  @Mock
+  private IntegrationUtils utils;
+
+  @Spy
+  private IntegrationProperties properties = new IntegrationProperties();
+
+  @InjectMocks
+  private KeyPairService keyPairService = new KeyPairService(arguments);
+
+  @Before
+  public void init() throws IOException {
+    PowerMockito.mockStatic(Runtime.class);
+    PowerMockito.mockStatic(FileSystems.class);
+    PowerMockito.mockStatic(Files.class);
+
+    FileSystem fileSystem = mock(FileSystem.class);
+
+    when(FileSystems.getDefault()).thenReturn(fileSystem);
+    when(Files.getFileAttributeView(any(Path.class), eq(PosixFileAttributeView.class),
+        eq(LinkOption.NOFOLLOW_LINKS))).thenReturn(attributeView);
+    when(Runtime.getRuntime()).thenReturn(runtime);
+
+    doReturn(lookupService).when(fileSystem).getUserPrincipalLookupService();
+
+    List<String> optionValues = Arrays.asList(Boolean.TRUE.toString());
+    doReturn(optionValues).when(arguments).getOptionValues(GENERATE_CERTIFICATE);
+
+    doReturn(process).when(runtime).exec(anyString());
+    doReturn(process).when(runtime).exec(any(String[].class));
+
+    doReturn(new ByteArrayInputStream(new byte[]{})).when(process).getErrorStream();
+
+    Certificate certificateInfo = new Certificate();
+    certificateInfo.setCaKeyPassword(MOCK_KEY_PASSWORD);
+    certificateInfo.setCaKeyFile(MOCK_KEY_FILE);
+    certificateInfo.setCaCertFile(MOCK_CERT_FILE);
+    certificateInfo.setCaCertChainFile(MOCK_CERT_CA_CHAIN_FILE);
+
+    this.properties.setSigningCert(certificateInfo);
+
+    doReturn(StringUtils.EMPTY).when(utils).getCertsDirectory();
+  }
+
+  @Test
+  public void testShouldNotGenerateCertificate() throws InterruptedException {
+    doReturn(Collections.EMPTY_LIST).when(arguments).getOptionValues(GENERATE_CERTIFICATE);
+
+    keyPairService.exportCertificate(null, null);
+
+    verify(process, times(0)).waitFor();
+  }
+
+  @Test(expected = KeyPairException.class)
+  public void testFailGenerateKeyPair() throws IOException {
+    Application application = new Application();
```

Comment:
Create common method for use in five locals for example:
   private Application getApplication() {
    Application application = new Application();
    application.setId(MOCK_APP_ID);
    application.setComponent(MOCK_APP_TYPE);
    return application;
  }